signing_clients/tests/__init__.py

```diff
@@ -97,9 +102,17 @@ def test_file(fname):
 
 class SigningTest(unittest.TestCase):
 
+    def setUp(self):
+        self.tmpdir = tempfile.mkdtemp()
```

Comment:
It would be better to add cleanup right under this line:

    self.addCleanup(lambda: shutil.rmtree(self.tmpdir))

because that will ensure cleanup even if you add more stuff to the setUp function that might raise an exception.